cdap-app-templates/cdap-etl/cdap-etl-realtime/src/main/java/co/cask/cdap/templates/etl/realtime/ETLWorker.java

```diff
@@ -80,16 +82,32 @@ public void initialize(final WorkerContext context) throws Exception {
     Preconditions.checkArgument(runtimeArgs.containsKey(Constants.Source.SPECIFICATION));
     Preconditions.checkArgument(runtimeArgs.containsKey(Constants.Sink.SPECIFICATION));
     Preconditions.checkArgument(runtimeArgs.containsKey(Constants.Transform.SPECIFICATIONS));
+    Preconditions.checkArgument(runtimeArgs.containsKey(Constants.Realtime.UNIQUE_ID));
 
     adapterName = runtimeArgs.get(Constants.ADAPTER_NAME);
```

Comment:
Then we will not know what records to clean. Should we handle that case (ie, delete the key corresponding to just "adapterName") or should we just let Preconditions throw an exception?